tests/test_collections.py

```diff
@@ -0,0 +1,72 @@
+import pytest
+from unittest import mock
+import os
+
+from tinydb import (
+    Query,
+    TinyDB,
+)
+from tinydb.storages import MemoryStorage
+
+from httlemon import collections
+
+
+class TestCollections:
+
+    class TestAdd:
+
+        def it_raises_invalid_collection_name_for_empty_name(self):
+
+            with pytest.raises(collections.InvalidCollectionName):
+                collections.add('')
+
+        def test_it_invalid_collection_name_for_non_string(self):
+
+            with pytest.raises(collections.InvalidCollectionName):
+                collections.add(1)
+
+        @mock.patch('httlemon.collections.TinyDB')
+        @mock.patch('httlemon.collections.DB_STORAGE')
+        @mock.patch('httlemon.collections.Path')
+        def test_it_inserts_a_new_collection(self, mock_path, mock_storage, mock_db):
+
+            mock_path.home.return_value = '/a_home_dir'
+            db = mock.Mock()
+            table = mock.Mock()
+            db.table.return_value = table
+            mock_db.return_value = db
+            storage = mock.Mock()
+            mock_storage.return_value = storage
+
+            collections.add('Lemon tree API')
+
+            mock_db.assert_called_once_with(
+                '/a_home_dir/httlemon/httlemon_db.json',
+                storage,
+            )
+            db.table.assert_called_once_with('collections')
+            table.insert.assert_called_once_with({'name': 'Lemon tree API'})
+
+class TestIntegration:
```

Comment:
These tests should have no `patch`s whatsoever and should be located in a different directory (maybe `tests/integration/test_collections.py` or `tests/functional/...`)? We could also make a simple script to run them in a docker container so that we don't mess with the user's laptop.